### ISSUE TITLE
flux-top: error decoding a job record from job-list RPC

### ISSUE THREAD

**Issue by grondo** [flux-framework/flux-core#5793]
While testing #5782 and using `flux top`, I often see this error:
```
flux-top: error decoding a job record from job-list RPC
```
which just causes `flux-top` to abort immediately.

I believe what is happening here is that a job which never had a `start` or `alloc` event can be in CLEANUP for a short time if it is canceled. In this case, the job will not have many of the fields which are assumed to be there since `flux top` ostensibly lists only "running" jobs. This is an artifact of being able to transition from DEPEND, PRIORITY, or SCHED states directly to CLEANUP on a fatal job exception, making it a bit awkward to consider these jobs RUNNING. However, in this case `flux top` can probably just skip these jobs instead of fatally aborting.

```
flux-top: error decoding a job record from job-list RPC: Object item not found: t_run: 
 {"id":560862330880,"userid":6885,"state":32,"name":"true","ntasks":1}
```

**Comment by chu11** [comment]
```
flux-top: error decoding a job record from job-list RPC: Object item not found: t_run: 
 {"id":560862330880,"userid":6885,"state":32,"name":"true","ntasks":1}
 ```
 
 I assume this was output w/ some extra debugging you added?
 
 It occurred to me that we call job-list requesting only running jobs

```
    if (!(f = flux_rpc_pack (joblist->top->h,                                                                                               
                             "job-list.list",                                                                                               
                             0,                                                                                                             
                             0,                                                                                                             
                             "{s:i s:{s:[i]} s:[s,s,s,s,s,s,s,s]}",                                                                         
                             "max_entries", win_dim.y_length - 1,                                                                           
                             "constraint",                                                                                                  
                             "states", FLUX_JOB_STATE_RUNNING,                                                                              
                             "attrs",                                                                                                       
                               "annotations",                                                                                               
                               "userid",                                                                                                    
                               "state",                                                                                                     
                               "name",                                                                                                      
                               "queue",                                                                                                     
                               "nnodes",                                                                                                    
                               "ntasks",                                                                                                    
                               "t_run"))                                                                                                    
        || flux_future_then (f, -1, joblist_continuation, joblist) < 0)                                                                     
        fatal (errno, "error sending job-list.list RPC request");    
```

so I'm not sure how this error is being hit.  Will look into `job-list` module and see if there's some potential error in there.

**Comment by grondo** [comment]
> I assume this was output w/ some extra debugging you added?

Yes, that was added debugging.

> so I'm not sure how this error is being hit.

My assumption is explained above. There is a weird case with Flux's definition of RUNNING, since a job is considered RUNNING in CLEANUP state, but a fatal job exception before a job starts could cause the job to go from SCHED->CLEANUP. Therefore a job that was never started can be considered RUNNING. 

I propose we just skip these jobs in `flux top`, but just haven't gotten around to proposing the patch.

**Comment by chu11** [comment]
> since a job is considered RUNNING in CLEANUP state

Ahhh I forgot that, now what you wrote above makes more sense.

**Comment by chu11** [comment]
Also, now that we have constraints in job-list, we could return only jobs with a t_run > 0.

**Comment by grondo** [comment]
That's a pretty good idea, though we do have to worry about version compatibility so it might be best to keep the query as simple as possible. It is pretty easy to just skip jobs that fail to unpack (the condition probably only occurs in the run/cancel storm test workload anyway)

**Comment by grondo** [comment]
Should have been closed by #5821

**Comment by chu11** [comment]
> Should have been closed by #5821

Oh crap, I completely cut and pasted the wrong issue number.  Sorry.

**Comment by grondo** [comment]
NP!
